src/circuits/noise.jl

```diff
@@ -0,0 +1,158 @@
+# Noise model gate definitions
+function gate(::GateName"1qubit-pauli_error"; pX::Number = 0.0, pY::Number = 0.0, pZ::Number = 0.0)
```

Comment:
Maybe `"1qubit_pauli_error"`? Though I guess it doesn't matter much since most people would just use `"pauli_error"` with `N = 1` or `2`.

Reply:
Swap this with underscored version

Reply:
Rename: ` _gate(::GateName"1qubit-pauli_error"; kwargs...)` to make clear it is internal.